subprojects/core/src/main/java/org/gradle/internal/service/scopes/WorkerSharedProjectScopeServices.java

```diff
@@ -154,7 +154,8 @@ DefaultProjectLayout createProjectLayout(FileResolver fileResolver, FileCollecti
     protected ProjectScopedCache createProjectScopedCache(TemporaryFileProvider temporaryFileProvider, CacheRepository cacheRepository) {
         return new DefaultProjectScopedCache(temporaryFileProvider.newTemporaryFile(".cache"), cacheRepository);
```

Comment:
I would expect that we instead make this call lazy as opposed to creating a second instance of a `DefaultProjectScopedCache` below. IMO a `DefaultProjectScopedCache`, and therefore a `AbstractScopedCache` should take a supplier for a root dir instead of having a concrete root dir upon construction. 


The `TemporaryFileProvider` we capture here _is_ lazy enough to handle the buildDir changing, so we just need to make our call to `temporaryFileProvider.newTemporaryFile(".cache")` more lazy as well.

Reply:
We discussed this.  We want to limit the scope of the change and avoid changing `AbstractScopedCache`.  We don't expect the potential second instance of `DefaultProjectScopedCache` to cause any issues.